infrastructure/k8s/base/api-deployment.yml

```
@@ -0,0 +1,142 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: api-deployment
+  namespace: ~
+  labels:
+    name: api
+spec:
+  strategy:
+    type: RollingUpdate
+    rollingUpdate:
+      maxUnavailable: 1
+      maxSurge: 1
+  replicas: 2
+  selector:
+    matchLabels:
+      component: api
+  template:
+    metadata:
+      labels:
+        component: api
+    spec:
+      containers:
+        - image: IMAGE_URL
+          name: api
+          ports:
+            - containerPort: 3000
+              name: api
+          livenessProbe:
+            httpGet:
+              path: /
+              port: 3000
+            initialDelaySeconds: 3
+            periodSeconds: 3
+          env:
```

```suggestion
          envFrom:
            - secretRef:
                name: api-secrets
```
no need to specify entry for each env variable if used like that
reference: [Leaderator - Deployment](https://github.com/SoftwareBrothers/leaderator-api/blob/master/k8s/base/api-deployment.yaml)

---

Thanks, I'll do it in this task.